ocs_ci/deployment/ibmcloud.py

```diff
@@ -111,3 +121,217 @@ def check_cluster_existence(self, cluster_name_prefix):
+        """
+        Retrieve release image using the openshift installer.
+        """
+        logger.info("Retrieving release image")
+        cmd = f"{self.ocp_deployment.installer} version"
+        proc = exec_cmd(cmd)
+        for line in proc.stdout.decode().split("\n"):
+            if "release image" in line:
+                return line.split(" ")[2].strip()
+
+    def get_resource_group(self):
+        """
+        Retrieve and set the resource group being utilized for the cluster assets.
+        """
+        cmd = "ibmcloud resource groups --output json"
+        proc = exec_cmd(cmd)
+        logger.info("Retrieving cluster resource group")
+        resource_data = json.loads(proc.stdout)
+        for group in resource_data:
+            if group["name"].startswith(self.cluster_name):
+                # TODO: error prone if cluster_name is a substring of another cluster
+                logger.info(f"Found resource group: {group['name']}")
+                return group["name"]
+        logger.info(f"No resource group found with cluster name: {self.cluster_name}")
+
+    def delete_service_id(self):
+        """
+        Delete the Service ID.
+        """
+        logger.info("Deleting service ID")
+        cmd = (
+            f"ccoctl ibmcloud delete-service-id --credentials-requests-dir {self.credentials_requests_dir} "
+            f"--name {self.cluster_name}"
+        )
+        exec_cmd(cmd)
+
+    def delete_volumes(self, resource_group):
+        """
+        Delete the pvc volumes created in IBM Cloud that the openshift installer doesn't remove.
+
+        Args:
+            resource_group: Resource group in IBM Cloud that contains the cluster resources.
+
+        """
+        if resource_group:
+            logger.info("Deleting volumes")
+            cmd = (
+                f"ibmcloud is vols --resource-group-name {resource_group} --output json"
+            )
+            proc = exec_cmd(cmd)
+
+            volume_data = json.loads(proc.stdout)
+            volume_ids = [volume["id"] for volume in volume_data]
+
+            if volume_ids:
+                cmd = f"ibmcloud is vold -f {' '.join(volume_ids)}"
+                exec_cmd(cmd)
+            else:
+                logger.info(f"No volumes found in resource group: {resource_group}")
+
+    def delete_resource_group(self, resource_group):
+        """
+        Delete the resource group that contained the cluster assets.
+
+        Args:
+            resource_group: Resource group in IBM Cloud that contains the cluster resources.
+
+        """
+
+        @retry(CommandFailed, tries=3, delay=30, backoff=1)
+        def _delete_group():
+            cmd = f"ibmcloud resource group-delete {resource_group} -f"
+            exec_cmd(cmd)
+
+        if resource_group:
+            logger.info(f"Deleting resource group: {resource_group}")
+            _delete_group()
+
+    @staticmethod
+    def check_cluster_existence(cluster_name_prefix):
```

Comment:
@clacroix12 , can this be a base class method ? Or is it different from managed cluster?

Reply:
The base class is `CloudDeploymentBase` and as such this is too specific to be included in that class. This has IBM specifics. You can see this also implemented in the `IBMCloud` class. I *could* create another base class these both pull from but in the interest of keeping things much simpler I chose to just implement this method in my new class.